app/controllers/sessions_controller.rb

```diff
@@ -17,7 +17,8 @@ def create
     if should_authorize_only
       set_session_auth
       flash[:notice] = "You are now authenticated as #{auth_hash["info"]["name"] || auth_hash["info"]["battletag"]} for the next 30 minutes."
-      redirect_to "#{request.base_url}/#{omniauth_params["redirect_path"].presence || ""}"
+      path = omniauth_params["redirect_path"].presence || ""
+      redirect_to "#{request.base_url}#{path.starts_with?("/") ? "" : "/"}#{path}"
```

Comment:
This is a little hard to read, it's also copied a few lines down. Would it be possible to move it to it's own method?